### ISSUE TITLE
Include the support for subMerchant information

### ISSUE THREAD

**Issue by laith-cko** [checkout/checkout-sdk-php#118]
### Environment
* Checkout SDK version: checkout-sdk-php/1.0.13 + latest version

### Description

Due to the recent compliance requirements in the region. Merchants would need to submit subMerchant information in the API request. Would you guys be able to add the support for object highlighted below in the PHP SDK? all the highlighted fields should be optional to allow other merchants to skip sending those parameters if not required in their region/agreement.
Here’s how the object should look like in a payment request:
{
 “source”: {
   “type”: “token”,
   “token”: “tok_hgt678iuhGtyuioiHgyu89”
 },
 “amount”: 4500,
 “currency”: “SAR”,
 “reference”: “Fyugtr56t678Yhui67",

 ```
"processing": {
   "aggregator": {
     "sub_merchant_id": "9874587412",
     "sub_merchant_name": "Foodics - catering business",
     "sub_merchant_legal_name": "Foodics catering service LLC",
     "sub_merchant_street": "Kuwait Street 1",
     "sub_merchant_city": "Kuwait City",
     "sub_merchant_country": "KWT",
     "sub_merchant_postal_code": "60000",
     "sub_merchant_state": "Kuwait",
     "sub_merchant_email": "[foodcs@support.com](mailto:foodcs@support.com)",
     "sub_merchant_phone": "[+965412478112](tel:+965412478112)",
     "sub_merchant_Industry_Code": "5411"
   }
 }
```

}

**Comment by Armando-Ibarra** [comment]
Hi @laith-cko .

Could you tell us for which version `NAS` or `ABC` does this request belongs to?

Does this `aggregator` property is a future release for all the SDK or just a for PHP?

**Comment by laith-cko** [comment]
Hi Armando. This is for ABC only. The object is not yet supported on NAS. 
This request is only for specific merchants so I don't think you will receive similar requests for the other SDKs. But yes all SDKs on ABC should ideally support this object, but what's urgent now is the PHP SDK. 

> The target of this change will based on the latest version 2.1.0
Acknowledged

**Comment by Armando-Ibarra** [comment]
Perfect, one last thing, is this property available to test in Sandbox, if so what are the requirements needed.

**Comment by laith-cko** [comment]
Yes it's available, no requirements from the integration perspective. 
I guess you're trying to document a guide for the merchant? we would need to enable something at our end in order for these details to be sent to the bank, but it doesn't need to be "allowed" from the integration perspective. 

Layth Mohammed
Director, Implementation Engineering

--- original message ---
On April 21, 2022, 12:27 AM GMT+4 ***@***.*** wrote:



Perfect, one last thing, is this property available to test in Sandbox, if so what are the requirements needed.

—

Reply to this email directly, view it on GitHub, or unsubscribe.

You are receiving this because you were mentioned.Message ID: ***@***.***>
--- end of original message ---
-- 


***************************************

This message is intended solely 
for the addressee and may contain confidential information. If you have 
received this message in error, please send it back to us, immediately and 
permanently delete it, and do not use, copy or disclose the information 
contained in this message or in any attachment. Any unauthorized use is 
strictly prohibited.



***************************************

**Comment by Armando-Ibarra** [comment]
Hi @laith-cko .
We released version [2.1.1](https://github.com/checkout/checkout-sdk-php/releases/tag/2.1.1) with the requested changes, please let us know what does the customer think.

Regards!
